notify/src/inotify.rs

```diff
@@ -296,6 +296,28 @@ impl EventLoop {
                                 );
                                 remove_watch_by_event(&path, &self.watches, &mut remove_watches);
                             }
+                            if event.mask.contains(EventMask::DELETE_SELF) {
+                                evs.push(
+                                    Event::new(EventKind::Remove(match path.clone() {
+                                        Some(path) => {
+                                            let current_watch = self.watches.get(&path);
+                                            match current_watch {
+                                                Some(&(_, _, _, is_dir)) => {
+                                                    if is_dir {
```

Comment:
this could also be collapsed into a single match over the cases Some(..,true,), Some(..,false), None

Reply:
Done